packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -443,6 +762,16 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
   @override
   Widget build(BuildContext context) {
     final List<Widget> children = <Widget>[];
+    if (_currentAnimation.status == AnimationStatus.completed && widget.child != null) { 
```

Comment:
What happens when the FAB is changed while it is animating to a new position? (or animating to a new position while the scale animation triggered by a FAB change is running?)

Looks like we might jump from between the different cases in this build method during an animation?

Reply:
Previously, you were right about the jump.  I'm combining the animations now, which avoids the issue of a jump between them.